### PR TITLE
Dynamic Simulation - Curve configuration for BUS

### PR DESCRIPTION
**Pull request by thangqp**
Support bus and bus bar section in expert filter

_**Related PRs**_

**Front-end :** 
https://github.com/gridsuite/gridstudy-app/pull/1763 (with demo)

**Back-end :**
https://github.com/gridsuite/dynamic-simulation-server/pull/79
https://github.com/gridsuite/study-server/pull/489

### CODE REVIEW

**Review by EstherDarkish — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/FilterService.java [hunk 1]
```diff
@@ -592,6 +593,25 @@ private List<Identifiable<?>> getVscConverterStationList(Network network, Abstra
...
+        if (filter instanceof ExpertFilter expertFilter) {
+            // topologyKind is an optional info attached into expert filter when filtering bus for optimizing the perf
+            // note that with voltage levels of kind TopologyKind.NODE_BREAKER, buses are computed on-the-fly => expensive
+            var topologyKind = expertFilter.getTopologyKind();
+            Predicate<VoltageLevel> voltageLevelFilter = vl -> topologyKind == null || vl.getTopologyKind() == topologyKind;
+
+            Stream<Identifiable<?>> stream = network.getVoltageLevelStream()
+                    .filter(voltageLevelFilter)
+                    .map(VoltageLevel::getBusBreakerView)
+                    .flatMap(VoltageLevel.BusBreakerView::getBusStream).map(bus -> bus);
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/main/java/org/gridsuite/filter/server/FilterService.java]: DONE! Redundant removed

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java [hunk 2]
```diff
@@ -16,33 +16,47 @@
  * @author Antoine Bouhours <antoine.bouhours at rte-france.com>
  */
 public final class ExpertFilterUtils {
+
+    public static final String FIELD_AND_TYPE_IS_NOT_IMPLEMENTED_WITH_EXPERT_FILTER = "These field and type are not implemented with expert filter";
+
+    public static final String TYPE_IS_NOT_IMPLEMENTED_WITH_EXPERT_FILTER = "This type is not implemented with expert filter";
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java]: DONE

File: src/test/java/org/gridsuite/filter/server/ExpertFilterUtilsTest.java [hunk 3]
```diff
@@ -362,4 +362,172 @@ void testEvaluateExpertFilterBetweenOperator() {
         Mockito.when(gen.getMaxP()).thenReturn(20.0);
         assertFalse(numFilter.evaluateRule(gen));
     }
+
+    @Test
+    void testEvaluateExpertFilterBusAndBusBarSectionWithInAndNotIn() {
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/test/java/org/gridsuite/filter/server/ExpertFilterUtilsTest.java:367]: A technical ticket will be created after merge in PR

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java [hunk 4]
```diff
@@ -16,33 +16,47 @@
  * @author Antoine Bouhours <antoine.bouhours at rte-france.com>
  */
 public final class ExpertFilterUtils {
+
+    public static final String FIELD_AND_TYPE_NOT_IMPLEMENTED = "These field and type are not implemented with expert filter";
```
[the remark this answers is not in the document]
Comment (thangqp) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java]: DONE by https://github.com/gridsuite/filter-server/pull/82/commits/05ac820d6ff5a6b0a5ff071cb02f173bbc1e19a0

[the remark this answers is not in the document]
Comment (flomillot) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java]: missing "combination"

**Review by flomillot — APPROVED** [review]

File: src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (thangqp) [reply src/main/java/org/gridsuite/filter/server/utils/expertfilter/ExpertFilterUtils.java]: DONE

**Pull request merged**